src/protocol/sort/generate_permutation.rs

```diff
@@ -62,9 +70,18 @@ pub(super) async fn shuffle_and_reveal_permutation<
     )
     .await?;
 
-    // TODO: THIS IS WHERE VALIDATOR WILL BE CALLED!
-    let revealed_permutation =
-        reveal_permutation(ctx.narrow(&RevealPermutation), &shuffled_permutation).await?;
+    let mut revealed_permutation = Vec::new();
+    if let Some(malicious_validator) = malicious_validator {
+        revealed_permutation = malicious_validator
+            .validate(ShuffledPermutationWrapper {
+                val: shuffled_permutation,
```

Comment:
@martinthomson
Error: expected struct `MaliciousReplicated`, found type parameter `S`
Is there a way to bind a generic S to a struct since, this code will only be called for MaliciousReplicated while shuffle_and_reveal_permutation takes Vec<S> as an input?